apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs

```diff
@@ -167,6 +132,29 @@ public UrlSigner UrlSigner
                 UnauthenticatedAccess = true
             }.Build();
 
+        /// <summary>
+        /// Creates a URL signer built base on this client, that is using the same credential as this client, as well
+        /// as defaulting to this client's URI scheme, host and port. If the credential used by this client is not compatible
+        /// with <see cref="UrlSigner"/> this method will throw <see cref="InvalidOperationException"/>.
+        /// See <see cref="UrlSigner"/>'s documentation for more information on compatible credentials.
+        /// </summary>
+        /// <remarks>
+        /// Because credentials used by this client may changed, this method will always return a new <see cref="UrlSigner"/> instance
+        /// each time it is called.
+        /// </remarks>
+        public UrlSigner CreateUrlSigner()
+        {
+            UrlSigner signer = UrlSigner.FromCredential(Service.HttpClient.MessageHandler.Credential);
+
+            var baseUri = new Uri(Service.BaseUri);
+            return signer.WithDefaultOptionsOverride(new UrlSigner.DefaultOptionsOverrides(
+                baseUri.Scheme, baseUri.Host,
+                // If the original URI didn't specify a port, we want signed URLs to not have a port either;
+                // but Uri.Port returns the default port for the URI scheme when the port was not specified on the
+                // original URI.
+                baseUri.IsDefaultPort && !Service.BaseUri.Contains(baseUri.Port.ToString()) ? null : baseUri.Port));
```

Comment:
I'd personally move this logic (including the comment) into a separate local variable declaration of `port`, so we can then just use `new UrlSigner.DefaultOptionsOverrides(baseUri.Scheme, baseUri.Host, port)`.